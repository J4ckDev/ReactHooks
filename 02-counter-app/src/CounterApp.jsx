// Snipet rafcp
// MÃ¡s eventos disponibles en React en https://es.reactjs.org/docs/events.html

import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);
  // Otra forma de incrementar
  // const handleAdd = () => setCounter( (c) => c + 1 );

  const handleReset = () => setCounter(value);

  const handleSubstract = () => setCounter(counter - 1);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button className="button left" onClick={ handleAdd }>+1</button>
      <button className="button middle" onClick={ handleReset }>Reset</button>
      <button className="button right" onClick={ handleSubstract }>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
    value: 0
}

export default CounterApp;
