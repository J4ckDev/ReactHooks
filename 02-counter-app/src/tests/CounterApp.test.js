import { shallow, ShallowWrapper } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas al componente CounterApp", () => {
  // Como las pruebas se hacen una seguida de la otra, se recomienda usar beforeEach para
  // reiniciar el componente al estado inicial

  let wrapper = shallow(<CounterApp />); // Esto es una mala practica y debería ser solo let wrapper; pero es para no perder el intellisense de VSCode

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("Debe mostrar <CounterApp /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe mostrar el valor por defecto de 100", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const counterValue = wrapper.find("h2").text().trim();
    expect(counterValue).toBe(`${value}`);
  });
  test("Debe incrementar el valor del contador con el boton +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const counterValue = wrapper.find("h2").text().trim();
    expect(counterValue).toBe("1");
  });
  test("Debe reducir el valor del contador con el boton -1", () => {
    wrapper.find("button").at(2).simulate("click");
    const counterValue = wrapper.find("h2").text().trim();
    expect(counterValue).toBe("-1");
  });
  test("Debe reiniciar el contador con el botón reset", () => {
    wrapper.find("button").at(2).simulate("click");
    wrapper.find("button").at(2).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const counterValue = wrapper.find("h2").text().trim();
    expect(counterValue).toBe("0");
  });
});
