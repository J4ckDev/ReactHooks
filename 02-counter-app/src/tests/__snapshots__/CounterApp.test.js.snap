// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente CounterApp Debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    className="button left"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="button middle"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="button right"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
